cmd/generate/cmd_util.go

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/nginxinc/nginx-go-crossplane/internal/generator"
+)
+
+type filterFlag struct {
+	filter map[string]struct{}
+}
+
+func (reader *filterFlag) Set(value string) error {
+	if reader.filter == nil {
+		reader.filter = make(map[string]struct{})
+	}
+	reader.filter[value] = struct{}{}
+	return nil
+}
+
+func (reader *filterFlag) String() string {
+	return fmt.Sprintf("%v", *reader)
+}
+
+type overrideItem struct {
+	directive string
+	masks     []generator.Mask
+}
+
+func (item *overrideItem) UnmarshalText(text []byte) error {
+	rawOverride := string(text)
+
+	// rawStr should follow the format: directive:bitmask00|bitmask01|...,bitmask10|bitmask11|...
+	directive, definition, found := strings.Cut(rawOverride, ":")
+	if !found {
+		return fmt.Errorf("colon not found, override=%s, ", rawOverride)
+	}
+	directive = strings.TrimSpace(directive)
+
+	item.directive = directive
+	if directive == "" {
+		return errors.New("directive name is empty")
```

Comment:
Include the override to communicate which one is being rejected?

```go
return fmt.Errorf("directive name is empty: override=%s", rawOverride)
```

I think I would do this for most errors where it would be helpful to see the specific input being rejected.

Reply:
We moved it to `Set` to simplify the errors in item.UnmarshalText
https://github.com/nginxinc/nginx-go-crossplane/pull/106#discussion_r1678144644
```go
func (ov *override) Set(value string) error {
	if *ov == nil {
		*ov = override{}
	}
	var item overrideItem
	err := item.UnmarshalText([]byte(value))
	if err != nil {
		return fmt.Errorf("invalid override %s:%w", value, err)
	}

	(*ov)[item.directive] = item.masks
	return nil
}
```